File: back-end/SiteSync/src/main/java/ma/ac/esi/sitesync/chantier/service/interfaces/ChantierService.java
package ma.ac.esi.sitesync.chantier.service.interfaces;
 
import ma.ac.esi.sitesync.chantier.dto.ChantierRequest;
import ma.ac.esi.sitesync.chantier.dto.ChantierResponse;
import ma.ac.esi.sitesync.chantier.model.StatutChantier;
import org.springframework.web.multipart.MultipartFile;
import java.util.List;
 
public interface ChantierService {
    ChantierResponse create(ChantierRequest request);
    ChantierResponse findById(String id);
    List<ChantierResponse> findAll();
    List<ChantierResponse> findByStatut(StatutChantier statut);
    List<ChantierResponse> findByResponsable(String responsableId);
    List<ChantierResponse> search(String query);
    List<ChantierResponse> findNear(double lng, double lat, double rayon);
    ChantierResponse update(String id, ChantierRequest request);
    ChantierResponse updateStatut(String id, StatutChantier statut);
    ChantierResponse updateAvancement(String id, int avancement);
    ChantierResponse addPhoto(String id, MultipartFile file);
    ChantierResponse addDocument(String id, MultipartFile file);
    ChantierResponse addMembre(String id, String membreId);
    void delete(String id);
}

File: mobile/src/screens/chantier/ChantierService.ts
import api from '../../services/api';
import { Chantier, ChantierRequest } from './types';

export const chantierService = {
  async getAll(statut?: string, search?: string): Promise<Chantier[]> {
    const res = await api.get('/chantier', { params: { statut, search } });
    
    return res.data.data;
  },
  async getById(id: string): Promise<Chantier> {
    const res = await api.get(`/chantier/${id}`);
    return res.data.data;
  },
  async create(data: ChantierRequest): Promise<Chantier> {
    const res = await api.post('/chantier', data);
    return res.data.data;
  },
  async updateAvancement(id: string, avancement: number): Promise<Chantier> {
    const res = await api.patch(`/chantier/${id}/avancement`, null, { params: { avancement } });
    return res.data.data;
  },
  async delete(id: string): Promise<void> {
    await api.delete(`/chantier/${id}`);
  },
  
}